### PR TITLE
Add support for GameMgr based Blue Spiral Games.

### PR DESCRIPTION
**Pull request by Hoikas**
Adds support for legacy Blue Spiral games (eg MOULa).

Depends on #1308, #1217.

The BlueSpiral Python is quite a bit simpler than the marker game Python, so I am willing to re-order the PRs such that the BlueSpiral commit comes before the marker commit, allowing this PR to be reviewed and merged before #1217. These three PRs are all one branch on my machine, FWIW.

This PR introduces a minor change to the BlueSpiral gameplay logic, which will be upstreamed to OU soon. On Cyan's variant of BlueSpiral, clicking the bahro door while a game is running is supposed to start a new game. On H'uru, clicking on the bahro door while a game is running is disabled. In this PR, clicking on the bahro door while a game is running causes the current game to end as if someone had pressed an incorrect cloth. This change was made because Cyan's restart game functionality is buggy, and clicking on the door many times in quick succession yields comical results.

This requires H-uru/moul-assets#219.

### CODE REVIEW

**Comment by Hoikas** [comment]
Suggestions about the CI failures would be appreciated.

**Comment by Hoikas** [comment]
Converting to draft status based on observations from today's private test.

**Comment by Hoikas** [comment]
All known problems fixed; we're ready for review.

File: Scripts/Python/xBlueSpiral.py [hunk 1]
```diff
@@ -98,394 +98,528 @@
...
+            clothList = list(range(kNumCloths))
+            random.shuffle(clothList)
+            ageSDL[SDLBSKey.value] = (" ".join((str(i) for i in clothList)),)
+            return clothList
+        return [int(i.strip()) for i in clothList.split(" ")]
+
+    @property
+    def isLocallyOwned(self) -> bool:
+        """Is the game owned by the local client? This can be distinct from whether or not the
+           PythonFileMod's SceneObject."""
```
Comment (dpogue) [inline Scripts/Python/xBlueSpiral.py:154]: This sentence feels incomplete

Comment (Hoikas) [reply Scripts/Python/xBlueSpiral.py]: Should be better now.

**Comment by Hoikas** [comment]
I've re-ordered the commits so that this PR no longer depends on #1217.

**Review by dpogue — APPROVED** [review]
LGTM

**Comment by Hoikas** [comment]
Rebased to fix merge conflicts.

**Pull request merged**